src/oled_display_node.cpp

```diff
@@ -775,148 +773,162 @@ void displayApiCallback(const oled_display_node::DisplayOutput::ConstPtr& msg)
+        ROS_INFO("%s: Listening on topic /%s for messages of type %s", THIS_NODE_NAME,
+        ROS_TOPIC_DISPLAY_NODE, "DisplayOutput" );
+    } else {
+        ROS_ERROR("%s: Display did not initialize properly and will not be used! ", THIS_NODE_NAME);
+        return 0;
     }
 
+    // Display topic and we then get callbacks for each message
+    ros::Subscriber sub = nh.subscribe(ROS_TOPIC_DISPLAY_NODE, 100, displayApiCallback);
 
-    // If there is a motor_power_active topic and we get the callback also show motor power status
-    if (g_motorPowerActive >= 0) {
-        ROS_DEBUG("%s motor power active is now %d", THIS_NODE_NAME, g_motorPowerActive);
+    // Battery_state topic and we then get callbacks for each message
+    ros::Subscriber sub2 = nh.subscribe("battery_state", 100, batteryStateApiCallback);
 
-        // If you change the text, use same number of chars so display does not move around on the line
-        std::stringstream powStream;
-	if (g_motorPowerActive > 0) {
-            powStream <<   " ON";
-        }else {
-            powStream <<   "OFF";
+    // Motor_power_active topic
+    ros::Subscriber sub3 = nh.subscribe("motor_power_active", 100, motorPowerActiveApiCallback);
+
+    // Firmware state topic
+    ros::Subscriber sub4 = nh.subscribe("firmware_version", 100, firmwareVersionCallback);
+
+    // We will refresh the lines from time to time in case IP addr has changed
+    ros::Rate loop_rate(0.5);
+    int32_t loopIdx = 0;
+
+    // mainloop:
+    while ((dispError == 0) && ros::ok())
+    {
+        loopIdx++;
+        hostname = getPopen("uname -n");
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_HOSTNAME, 0, DISP_TEXT_START_MODE, hostname.c_str());
+        ros::Duration(updateDelay).sleep();
+
+        getIpAddressses(displayedETHAddress, displayedWLANAddress, 0);
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_IP_WLAN, 0, DISP_TEXT_START_MODE, displayedWLANAddress.c_str());
+        ros::Duration(updateDelay).sleep();
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_IP_ETH, 0, DISP_TEXT_START_MODE, displayedETHAddress.c_str());
+        ros::Duration(updateDelay).sleep();
+
+        // If there is a battery_state topic and we get the callback also show battery voltage
+        if (g_batteryVoltage > 0.0) {
+            ROS_DEBUG("%s Battery voltage is now %4.1f volts.", THIS_NODE_NAME, g_batteryVoltage);
+
+            // Control what shows up after the voltage reading.
+            // If you change the text, use same number of chars so display does not move around on the line
+            std::stringstream stream;
+            stream << std::fixed << std::setprecision(1) << g_batteryVoltage;
+
+            if (g_batteryVoltage <= BAT_LOW_LEVEL && (loopIdx % 2) == 0){
+                stream << "V  LOW";
+            }
+            else if (g_batteryVoltage >= BAT_CHARGING_LEVEL && (loopIdx % 3) != 0){
```

Comment:
This looks nice now.

Reply:
Yeah I've switched it to sometimes flash the percentage and always show CHRG instead of the opposite, since it's probably more in line with what's intended to be shown.